csharp/src/Ice/MetricsAdmin.cs

```diff
@@ -666,41 +607,47 @@ public static void ValidateProperties(string prefix, Communicator communicator)
             }
         }
 
-        private class MetricsMapFactory<T> : IMetricsMapFactory where T : Metrics, new()
+        private bool AddOrUpdateMap(string mapName, IMetricsMapFactory factory)
         {
-            public MetricsMapFactory(Action updater) => _updater = updater;
-
-            public void Update()
+            bool updated = false;
+            foreach (MetricsView v in _views.Values)
             {
-                Debug.Assert(_updater != null);
-                _updater();
+                updated |= v.AddOrUpdateMap(_communicator, mapName, factory, Logger);
             }
+            return updated;
+        }
 
-            public IMetricsMap Create(string mapPrefix, Communicator communicator) =>
-                new MetricsMap<T>(mapPrefix, communicator, _subMaps);
-
-            public void RegisterSubMap<S>(string subMap, System.Reflection.FieldInfo field)
-                where S : Metrics, new() => _subMaps.Add(subMap, new SubMapFactory<S>(field));
-
-            private readonly Action _updater;
-            private readonly Dictionary<string, ISubMapFactory> _subMaps = new Dictionary<string, ISubMapFactory>();
+        private MetricsView? GetMetricsView(string name)
```

Comment:
We could go for a one-liner `=> _views.TryGetValue(name, out MetricsView? view) ? view : (_disableViews.Contains(name) ? null : throw new UnknownMetricsView());`

Reply:
fixed